src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -111,28 +112,24 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
    * and return the value for storage in the symbol store
    */
   def staticDeclarationsRecursive(userInputs: WorkflowCoercedInputs, wdlFunctions: WdlStandardLibraryFunctions): Try[WorkflowCoercedInputs] = {
-    import scala.collection.mutable
-    val collected = mutable.Map[String, WdlValue]()
-    val allDeclarations = workflow.declarations ++ workflow.calls.flatMap {_.task.declarations}
-
-    val evaluatedDeclarations = allDeclarations.filter {_.expression.isDefined}.map {decl =>
-      val value = decl.expression.get.evaluate(declarationLookupFunction(decl, collected.toMap ++ userInputs), wdlFunctions)
-      collected += (decl.fullyQualifiedName -> value.get)
-      val coercedValue = value match {
-        case Success(s) => decl.wdlType.coerceRawValue(s)
-        case f => f
-      }
-      decl.fullyQualifiedName -> coercedValue
-    }.toMap
 
-    val (successes, failures) = evaluatedDeclarations.partition {case (_, tryValue) => tryValue.isSuccess}
-    if (failures.isEmpty) {
-      Success(successes.map {case (k,v) => k -> v.get})
-    } else {
-      val errors = failures.values.collect { case f: Failure[_] => f.exception.getMessage }
-        // .get because failures is guaranteed to be nonEmpty
-      Failure(new ValidationException("Could not evaluate some declaration expressions", errors.toList.toNel.get))
+    def evalDeclaration(accumulated: Map[String, Try[WdlValue]], current: ScopedDeclaration): Map[String, Try[WdlValue]] = {
+      current.expression match {
+        case Some(expr) =>
+          val successfulAccumulated = accumulated.collect({case (k, v) if v.isSuccess => k -> v.get })
+          val value = expr.evaluate(declarationLookupFunction(current, successfulAccumulated ++ userInputs), wdlFunctions)
+          accumulated + (current.fullyQualifiedName -> value)
+        case None => accumulated
+      }
     }
+
+    val workflowStaticDeclarations = workflow.scopedDeclarations.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)
+
+    val callStaticDeclarations = workflow.calls flatMap { call =>
+      call.scopedDeclarations.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)
+    } toMap
+
+    TryUtil.sequenceMap(workflowStaticDeclarations ++ callStaticDeclarations)
```

Comment:
@scottfrazer 

* I don't view one as being more arcane than the other. I think you're somewhat biased in that you already know the wizardry of one and not the other. If someone knew neither IMO they'd both look freaking weird.
* I have been surprised myself that they don't have a convenience implicit (at least not that I've found) for converting - I still don't fully believe it doesn't exist but I'll admit I've come across the same thing. IMO *that* is the sort of thing which should be in TryUtil. 
* Yes it is. My complaint about our homebrew scheme is that it's effectively equivalent plus some extra unboxing and doesn't provide any extra benefit AFAICT. That doesn't scream to me "please let's homebrew our own thing". So I'm not stating superiority, I'm saying that the tryutil scheme is *at best* equal and we shouldn't be blazing our own trail in cases where we're not providing a demonstrable value add.
* It spiders to exactly the same extent that Try does, which is to say as far as you want it to go. With the exception of "how do you operate over multiple of these" there really is no difference at all between how you handle a validation vs a try.

The one place where I think dealing with the validations is a bit of a headache is when you want to combine fail-slow with fail-fast. However it's still no different than what you'd have in the tryutil model, it's just different syntactically. The headache has been remembering what's intended to be fast vs slow and not combining them. At least in this model the two bits are differently typed (a validation vs a disjunction) so it's harder to screw it up w/o realizing it.